src/app/applications/multicast/multicast-table/multicast-table.component.ts

```diff
@@ -0,0 +1,134 @@
+import { DeleteDialogService } from '@shared/components/delete-dialog/delete-dialog.service';
+import { tableSorter } from '@shared/helpers/table-sorting.helper';
+import { MeService } from '@shared/services/me.service';
+import { SnackService } from '@shared/services/snack.service';
+import { Subscription } from 'rxjs';
+import { Multicast, MulticastData } from '../multicast.model';
+import { MulticastService } from '../multicast.service';
+
+@Component({
+  selector: 'app-multicast-table',
+  templateUrl: './multicast-table.component.html',
+  styleUrls: ['./multicast-table.component.scss'],
+})
+export class MulticastTableComponent
+  implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+  displayedColumns: string[] = ['groupName', 'groupType', 'menu'];
+  dataSource = new MatTableDataSource<Multicast>();
+  multicasts: Multicast[];
+  resultsLength = 0
+  public canEdit = false;
+  @Input() isLoadingResults: boolean = true;
+  public pageSize = environment.tablePageSize;
+  @Input() pageLimit: number;
+  public pageOffset = 0;
+  public pageTotal: number;
+  public applicationId: number;
+
+  private multicastSubscription: Subscription;
+  private deleteDialogSubscription: Subscription;
+
+  constructor(
+    private route: ActivatedRoute,
+    private deleteDialogService: DeleteDialogService,
+    private multicastService: MulticastService,
+    private meService: MeService,
+    public translate: TranslateService,
+    public snackService: SnackService
+  ) {
+    translate.use('da');
+  }
+
+  ngOnInit(): void {
+    this.applicationId = +Number(this.route.parent.snapshot.paramMap.get('id'));
+    this.getMulticasts(); // loads the multicasts
+    this.canEdit = this.meService.canWriteInTargetOrganization();
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+  }
+
+  getMulticasts(): void {
+    if (this.applicationId) {
+      this.multicastSubscription = this.multicastService
+        .getMulticastsByApplicationId(
+          // gets multicasts from db by applicationId
+          this.pageLimit,
+          this.pageOffset * this.pageLimit,
+          this.applicationId
+        )
+        .subscribe((multicasts: MulticastData) => {
+          this.multicasts = multicasts.data;
+          this.dataSource = new MatTableDataSource<Multicast>(this.multicasts); // these lines of code is inspired/taken from datatarget.
+          this.dataSource.paginator = this.paginator;
+          this.dataSource.sort = this.sort;
+          this.dataSource.sortingDataAccessor = tableSorter;
+          this.isLoadingResults = false;
+          if (this.pageLimit) {
+            this.pageTotal = Math.ceil(multicasts.count / this.pageLimit);
+          }
+          if (multicasts.ok === false) {
+            this.showLoadFailSnack(); // if not possible to load the multicasts, show error.
```

Comment:
Unnecessary comment; it's implicit. Same goes for most of the comments in this file